backend/b2/b2.go

```diff
@@ -629,6 +629,10 @@ func (f *Fs) list(ctx context.Context, dir string, recurse bool, prefix string,
 		if err != nil {
 			return err
 		}
+		// Check that there are matching files
+		if len(response.Files) == 0 {
```

Comment:
I'm unsure about this...

It potentially seems like a good idea (and one which could go in all the Bucket based remotes), however will it break stuff?

How does `rclone mount` behave with this patch if you make a new directory then list it for b2?

If we want all the bucket based remotes to behave like this then we should add an another integration test in fstest/fstests/fstests.go to list a non existent directory and make sure we get an error.



https://github.com/rclone/rclone/issues/3427

Reply:
i have never used `rclone mount` so i am not sure about its behavior, will test later today.

as i have said before, i think the bucket based remotes should either support empty folders all the way (via hidden files, as B2 and S3 already do) or not support them at all (`ErrorDirNotFound`).
the current way of appearing to support but silently not supporting is leading to a lot of confusion. ( as seen in #3427 & #3130)

whichever way you decide to go will need integration with the rest of rclone. this might need some broader fixing. (mount & --create-empty-src-dirs to start)

i can back out that commit for now and open it separately for a general PR to fix BBR whichever way you decide. however please merge the test & doc change.
also the current bugs/limitations docs imply that rclone does not create empty dirs, while to the user it appears that is does.

please let me know what you think

Reply:
i ran a brief test with `mount -vv` and besides for debug errors (which i also got for `.Trash` and other system files my file browser was looking for) it seemed to work fine, created and removed directory, add and delete files